src/engine/Server.cpp

```diff
@@ -656,31 +656,39 @@ boost::asio::awaitable<void> Server::processQuery(
     // Common code for sending responses for the streamable media types
     // (tsv, csv, octet-stream, turtle).
     auto sendStreamableResponse = [&](MediaType mediaType) -> Awaitable<void> {
-      auto responseGenerator = co_await computeInNewThread(
-          [&plannedQuery, &qet, mediaType, &cancellationHandle] {
-            return ExportQueryExecutionTrees::computeResultAsStream(
-                plannedQuery.value().parsedQuery_, qet, mediaType,
-                cancellationHandle);
-          },
+      auto responseGenerator = ExportQueryExecutionTrees::computeResultAsStream(
```

Comment:
Factor the generation of this (now rather lengthy) lambda into a separate function.